integration/nwo/fabric/network/network.go

```diff
@@ -259,8 +260,46 @@ func (n *Network) DeployChaincode(chaincode *topology.ChannelChaincode) {
 	if chaincode.Chaincode.InitRequired {
 		InitChaincode(n, chaincode.Channel, orderer, &chaincode.Chaincode, peers...)
 	}
+	//add new chaincode to the topology
+	n.topology.AddChaincode(chaincode)
```

Comment:
why do we need this/

Reply:
Because after the chaincode upgrade, topology has to show the new version of the chaincode.